### PR TITLE
Added plugin documentation

### PR DESCRIPTION
**Pull request by omercnet**
In order to load all the plugins we have to install the
dependencies and make integration_utils and base_service
available for import

can review the compiled version of this branch here http://honeycomb.cymmetria.com/projects/honeycomb-plugins/en/feature-plugin_docs/

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -15,7 +15,46 @@
...
+sys.path.append(os.path.abspath('..'))
+
 import honeycomb
+import services
+import integrations
+
+from honeycomb.servicemanager import base_service
+from honeycomb.integrationmanager import integration_utils
+
+services_list = next(os.walk('../services'))[1]
```
Comment (Sleeptime17) [inline docs/conf.py]: so we're definitely not doing "" anymore?

File: docs/conf.py [hunk 2]
```diff
@@ -15,7 +15,46 @@
...
+sys.path.append(os.path.dirname(honeycomb.__file__))
+sys.path.append(os.path.dirname(base_service.__file__))
+sys.path.append(os.path.dirname(integration_utils.__file__))
+
+for _ in services_list:
+    importlib.import_module('services.{}'.format(_))
+for _ in integrations_list:
+    importlib.import_module('integrations.{}'.format(_))
 
 project = u'Honeycomb Plugins'
```
Comment (Sleeptime17) [inline docs/conf.py]: Definitely not doing the ""?????

File: docs/conf.py [hunk 3]
```diff
@@ -15,7 +15,46 @@
...
+
+def install_plugins(dir, plugins, target):
+    for plugin in plugins:
+        reqs = os.path.join(dir, plugin, "requirements.txt")
+        if os.path.exists(reqs):
+            pipargs = ["install", "-r", reqs, "--target", target, "--ignore-installed"]
+            subprocess.check_call([sys.executable, "-m", "pip"] + pipargs)
+
+
+venv = tempfile.mkdtemp()
```
Comment (Sleeptime17) [inline docs/conf.py]: This doesn't look like plugin documentation.

Regardless, wouldn't you prefer creating your venv tempdir under a specific named subdir?

File: docs/conf.py [hunk 4]
```diff
@@ -15,7 +15,46 @@
...
+install_plugins('../services', services_list, venv)
+install_plugins('../integrations', integrations_list, venv)
+
+sys.path.append(venv)
+sys.path.append(os.path.dirname(honeycomb.__file__))
+sys.path.append(os.path.dirname(base_service.__file__))
+sys.path.append(os.path.dirname(integration_utils.__file__))
+
+for _ in services_list:
+    importlib.import_module('services.{}'.format(_))
```
Comment (Sleeptime17) [inline docs/conf.py]: It's worth it to catch import errors here, just in case. And 2 lines under.

File: docs/conf.py (hunk 3 above)
Comment (omercnet) [reply docs/conf.py]: nah it's just for sphinx build, doesn't matter

File: docs/conf.py (hunk 4 above)
Comment (omercnet) [reply docs/conf.py]: if import doesn't work than sphinx won't be able to generate the document, so it's better to break and alert us something is wrong
again this is only used during readthedocs build

File: docs/conf.py (hunk 1 above)
Comment (omercnet) [reply docs/conf.py]: I didn't actually think about it at all..  most of this file uses `'` so I'll stick to that

**Comment by omercnet** [comment]
@Sleeptime17 any comment on the documentation itself? `conf.py` is less important

**Comment by omercnet** [comment]
where did you see the ¶ symbol?

the content is coming from the code, so it should be fixed in a separate PR
I was referring more to the layout but if nothing stands out I guess it's all good :)

**Pull request merged**